lib/web/apiserver.go

```diff
@@ -338,13 +338,17 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 	h.POST("/webapi/github/login/console", httplib.MakeHandler(h.githubLoginConsole))
 
 	// U2F related APIs
+	// DELETE IN 9.x, superseded by /mfa/ endpoints (codingllama)
 	h.GET("/webapi/u2f/signuptokens/:token", httplib.MakeHandler(h.u2fRegisterRequest))
 	h.POST("/webapi/u2f/password/changerequest", h.WithAuth(h.u2fChangePasswordRequest))
-	h.POST("/webapi/u2f/signrequest", httplib.MakeHandler(h.mfaChallengeRequest))
-	h.POST("/webapi/u2f/sessions", httplib.MakeHandler(h.createSessionWithU2FSignResponse))
-	h.POST("/webapi/u2f/certs", httplib.MakeHandler(h.createSSHCertWithMFAChallengeResponse))
+	h.POST("/webapi/u2f/signrequest", httplib.MakeHandler(h.mfaLoginBegin))
+	h.POST("/webapi/u2f/sessions", httplib.MakeHandler(h.mfaLoginFinishSession))
+	h.POST("/webapi/u2f/certs", httplib.MakeHandler(h.mfaLoginFinish))
 
 	// MFA related endpoints
+	h.POST("/webapi/mfa/login/begin", httplib.MakeHandler(h.mfaLoginBegin))
+	h.POST("/webapi/mfa/login/finish", httplib.MakeHandler(h.mfaLoginFinish))
+	h.POST("/webapi/mfa/login/finishsession", httplib.MakeHandler(h.mfaLoginFinishSession))
```

Comment:
Sorry just looking at this, but IIRC our HTTP server is deprecated. All new handles should be a part of GRPC API. Do these have to be HTTP?

Reply:
These are meant for login, that's why we are keeping the HTTP endpoints. I describe them in the [WebAuthn RFD](https://github.com/gravitational/teleport/pull/7808); I believe that, by and large, reviewers are OK with the proposal.

I've also raised https://github.com/gravitational/teleport/issues/7966 to eventually move login to gRPC. I intent to follow up, but I didn't want to change too much at the same time for various reasons (including the fact that I'm still getting to know the codebase better).

Reply:
Oh this is lib/web, this is fine then.